File: cs11/lab8/GCD.c
//GCD.c
//Kevin Lee
//1480757
//lab8
//Finds the greatest common denominator of two numbers.
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

int main(){
   int i, n;
   char str[100];
   int x[2];

        printf("Enter a positive integer: ");
        for(i=0; i<2; i++){
           n= scanf(" %s", str);
        while( 1 ){
         sscanf(str, "%d", &x[i]);
         if( x[i] > 0){
            printf("Enter another positive integer: ");
            break;
         }
         printf("Please enter a positive integer: ");
         n = scanf(" %s", str);
      }
   }
int a=x[0];
int b=x[1];
int temp = (x[0] % x[1]);
        if(a > b){
            if(temp == 0) {
                printf("The GCD of %d and %d is %d",x[0],x[1],b);
            } else {
                while(temp > 0){
                    temp = (a % b);
                    a = b;
                    b = temp;
                    if(temp == 0) {
                        printf("The GCD of  %d   and   %d  is %d",x[0],x[1],a);
                    }
                }
            }
        }else {
            temp = (b % a);
            if(temp == 0) {
                printf("The GCD of  %d   and   %d   is %d",x[0],x[1],a);
            } else {
                while(temp > 0) {
                    temp = (b % a);
                    b = a;
                    a = temp;
                    if(temp == 0){
                        printf("The GCD of %d and %d is %d",x[0],x[1],b);
                    }
                }
            }


 return 0;
}
}

File: cs11/pa3/GCD.java
// GCD.java
// Kevin Lee
// 1480757
// pa3
// Finds the greatest common denominator of two numbers.

 import java.util.Scanner;

 class GCD{
	 public static void main( String[] args ){
		 int a,b,temp;
		 String GCD;
		 Scanner sc = new Scanner(System.in);
		 System.out.print("Enter a positive integer: ");

		 while(true){
		 while( !sc.hasNextInt() ){
		 sc.next();
		 System.out.print("Please enter a positive integer: "); 
		 }	
		 a=sc.nextInt();
		 if( a>0 ) break;
		 System.out.print("Please enter a positive integer: ");
		 }

		 System.out.print("Enter another positive integer: ");

		 while(true){
		 while( !sc.hasNextInt() ){
		 sc.next();
		 System.out.print("Please enter a positive integer: ");
		 }
		 b=sc.nextInt();
		 if(b>0 ) break;
		 System.out.print("Please enter a positive integer: ");
		 }

		 GCD = "The GCD of " + a + " and " + b + " is "; 
		 temp = a % b; 
		

		 if(a > b){
		 if(temp == 0) {
		 System.out.println(GCD + b);
		 } else {
		 while(temp > 0){
		 temp = a % b;
		 a = b;
		 b = temp;
		 if(temp == 0) {
		 System.out.println(GCD + a); }
		 }
   		 }
		 }else {
		 temp = b % a;
		 if(temp == 0) {
		 System.out.println(GCD + a);
		 } else {
		 while(temp > 0) {
		 temp = b % a;
		 b = a;
		 a = temp;
		 if(temp == 0){
		 System.out.println(GCD + b); }
		 }
              }
          }
      }
 }
